src/shogun/evaluation/CrossValidationWrapper.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+
+#ifndef CROSSVALIDATATION_WRAPPER_H__
+#define CROSSVALIDATATION_WRAPPER_H__
+
+#include <vector>
+#include <shogun/machine/Machine.h>
+#include <type_traits>
+#include <string_view>
+#include <memory>
+#include <queue>
+#include <utility>
+#include <shogun/evaluation/CrossValidation.h>
+
+#define IGNORE_IN_CLASSLIST
+
+namespace shogun
+{
+    template<typename MachineType>
+    class CrossValidationWrapper : public SGObject
+    {   
+    public:
+        CrossValidationWrapper(std::vector<std::pair<std::string_view, std::vector<float64_t>>> params, 
```

Comment:
I wonder whether fixing the type here to float64 is best?
@gf712 what do you think?

Reply:
Hmm we can use the widest representation, and then it cast down when passed to the setters? Otherwise this could be Any..

Reply:
What do you mean exactly?

Reply:
I mean either we use a std::vector<Any> for each parameter space, or represent all parameters as `float64`, since then we can convert that to any other type, like `int64`. We could template this, but not sure if this has to be exposed to interfaces?

Reply:
ah I see. I think probably it is best to keep the interface simple here. @LiuYuHui maybe you can add a test/use-case/example for a float parameter and an int parameter (where the float is converted, and a runtime error occurs if the float cannot be casted to an int in a lossless manner)